### PR TITLE
Refactored makeRoomForBlock.

### PR DESCRIPTION
**Pull request by zuyu**
This refactor includes:
 * Renamed to `makeRoomForBlockOrBlob`.
 * Removed a unnecessary variable.
 * Simplified the `if-else` statement.
 * Added more comments about this method.

### CODE REVIEW

**Comment by cramja** [comment]
@zuyu did we run the TPCH query 8/9 on this?

**Comment by cramja** [comment]
@zuyu Also, when thinking about refactoring all the method names in EvictionPolicy.hpp from `blockPastTenseVerb` to `blockOrBlobPastTenseVerb` I think it seems too wordy.

Maybe we could make the naming more succinct to instead name them all of the form `notifyPastTenseVerb(block_id block_or_blob)`.

What do you think?

**Comment by cramja** [comment]
BTW `pastTenseVerb` is one of `{created, evicted, unreferenced, referenced, deleted}`

**Comment by zuyu** [comment]
@cramja I did not run TPC-H over this PR. This just refactors a method internally.

**Comment by zuyu** [comment]
@cramja Alternatively for `EvictionPolicy`, we could add a comment before the class definition: `block refers to a block or blob.`

**Comment by cramja** [comment]
@zuyu I think the simplest thing here is to use you approach. I will make a PR for this.

**Comment by pateljm** [comment]
Merging.
